diff --git a/lib/publish.js b/lib/publish.js
index 838c654d247f23f0c9de4f1594d877e6b4db8c43..cdc485acbf45578eac0b124604d3b5e155c92667 100644
--- a/lib/publish.js
+++ b/lib/publish.js
@@ -53,9 +53,10 @@ export default async function publish(pluginConfig, context, { Octokit }) {
     name: template(releaseNameTemplate)(context),
     body: template(releaseBodyTemplate)(context),
     prerelease: isPrerelease(branch),
-    make_latest: isLatestRelease(branch),
+    make_latest: 'legacy', // isLatestRelease(branch),
   };
 
+  debug("branch object: %O", branch);
   debug("release object: %O", release);
 
   const draftReleaseOptions = { ...release, draft: true };
