diff --git a/lib/publish.js b/lib/publish.js
index 838c654d247f23f0c9de4f1594d877e6b4db8c43..ed220c4de5f46dd254cab870a87137e6d5681a99 100644
--- a/lib/publish.js
+++ b/lib/publish.js
@@ -56,6 +56,7 @@ export default async function publish(pluginConfig, context, { Octokit }) {
     make_latest: isLatestRelease(branch),
   };
 
+  debug("branch object: %O", branch);
   debug("release object: %O", release);
 
   const draftReleaseOptions = { ...release, draft: true };
@@ -167,6 +168,7 @@ export default async function publish(pluginConfig, context, { Octokit }) {
     repo,
     release_id: releaseId,
     draft: false,
+    make_latest: isLatestRelease(branch),
   };
 
   // add discussion_category_name if discussionCategoryName is not undefined or false
